package.json

```diff
@@ -2,7 +2,9 @@
   "name": "@infinite-loop-factory/app-factory",
   "private": true,
   "scripts": {
+    "biome": "npx biome check --fix",
```

Comment:
P1


pnpm biome (@biomejs/biome) 의 커맨드랑 겹칠것 같아서 

lint-staged.config.mjs

혹시 여기에 어떤 커맨드에서 오류가 날까요?? 거기에 --unsafe 를 덧붙혀주면 해결되면서 커밋이 될것 같아요.

```js
    "biome check --files-ignore-unknown=true", // Check formatting and lint
    "biome check --write --unsafe --no-errors-on-unmatched", // Format, sort imports, lint, and apply safe fixes
    "biome check --write --unsafe --organize-imports-enabled=false --no-errors-on-unmatched", // format and apply safe fixes
    "biome check --write --unsafe --no-errors-on-unmatched", // Format, sort imports, lints, apply safe/unsafe fixes
```

Reply:
``` bash
[STARTED] Preparing lint-staged...
[COMPLETED] Preparing lint-staged...
[STARTED] Running tasks for staged files...
[STARTED] lint-staged.config.mjs — 41 files
[STARTED] *.{js,ts,cjs,mjs,d.cts,d.mts,jsx,tsx,json,jsonc} — 27 files
[STARTED] * — 41 files
[STARTED] package.json — 1 file
[STARTED] biome check --files-ignore-unknown=true
[STARTED] biome check --no-errors-on-unmatched --files-ignore-unknown=true
[STARTED] sort-package-json
********************************************************************
***[FAILED] biome check --no-errors-on-unmatched --files-ignore-unknown=true [FAILED]
***[FAILED] biome check --no-errors-on-unmatched --files-ignore-unknown=true [FAILED]
********************************************************************
[COMPLETED] Running tasks for staged files...
[STARTED] Applying modifications from tasks...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
********************************************************************
***[FAILED] biome check --files-ignore-unknown=true [FAILED]
***[FAILED] biome check --files-ignore-unknown=true [FAILED]
********************************************************************
[COMPLETED] Reverting to original state because of errors...
[STARTED] Cleaning up temporary files...
[COMPLETED] Cleaning up temporary files...
× biome check --no-errors-on-unmatched --files-ignore-unknown=true:
C:/Users/UserK/IdeaProjects/app-factory/apps/gd/tsconfig.json format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Formatter would have printed the following content:
```

```bash
"biome check  --fix --files-ignore-unknown=true", // Check formatting and lint
"biome check --no-errors-on-unmatched --files-ignore-unknown=true", // Check formatting and lint
```
위에 두 명령어고, **unsafe**를 붙여도 실패 나네요 ㅠㅠ
    
```diff
- biome check --files-ignore-unknown=true
+ biome check  --fix --files-ignore-unknown=true

- biome check --no-errors-on-unmatched --files-ignore-unknown=true
+ biome check --fix --no-errors-on-unmatched --files-ignore-unknown=true
```

--fix를 붙이면 성공하긴 하는데 fix를 넣으면 코드가 변경되니까 의도치 않게 안 넣으신 걸까요..?

Reply:
```

--write — Writes safe fixes, formatting and import sorting

--unsafe — Allow to do unsafe fixes, should be used with --write or --fix

--fix — Alias for --write, writes safe fixes, formatting and import sorting
```

아!! [biomejs 문서](https://biomejs.dev/reference/cli/)에 write랑 fix랑 같다고 되어있어서 write를 썼었는데 동작이 다른가보네요ㅠㅠ fix로 다 바꿔주셔도 될 것 같아요 감사합니다.

Reply:
eslint 쓸 때도 fix니 --write 버리고 다 --fix가 좋아보이네요😂

Reply:
저도 커밋할 때 규범님과 같은 현상이 재현되었는데

규범님이 올려주신대로
```diff
- biome check --files-ignore-unknown=true
+ biome check  --fix --files-ignore-unknown=true

- biome check --no-errors-on-unmatched --files-ignore-unknown=true
+ biome check --fix --no-errors-on-unmatched --files-ignore-unknown=true
```
`lint-staged.config.mjs`를 이렇게 수정하면 biome rule대로 포매팅 적용되어 커밋되네요!